mailet/standard/src/test/java/org/apache/james/transport/matchers/PartHasContentTypeTest.java

```diff
@@ -0,0 +1,105 @@
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Collection;
+
+import org.apache.james.core.MailAddress;
+import org.apache.james.core.builder.MimeMessageBuilder;
+import org.apache.mailet.Mail;
+import org.apache.mailet.base.test.FakeMail;
+import org.apache.mailet.base.test.FakeMatcherConfig;
+import org.junit.jupiter.api.Test;
+
+class PartHasContentTypeTest {
+    @Test
+    void shouldNotMatchWhenNoContentType() throws Exception {
+        PartHasContentType testee = new PartHasContentType();
+        testee.init(FakeMatcherConfig.builder()
+            .matcherName("PartHasContentType")
+            .condition("image/png")
+            .build());
+
+        Mail mail = FakeMail.builder()
+            .name("mail")
+            .recipient(ANY_AT_JAMES)
+            .mimeMessage(MimeMessageBuilder.mimeMessageBuilder()
+                .setMultipartWithBodyParts(
+                    MimeMessageBuilder.bodyPartBuilder()
+                        .disposition("attachment")
+                        .filename("xxx.zip")))
+            .build();
+        Collection<MailAddress> matched = testee.match(mail);
+
+        assertThat(matched)
+            .isNull();
+    }
+
+    @Test
+    void shouldNotMatchWhenWrongContentType() throws Exception {
+        PartHasContentType testee = new PartHasContentType();
+        testee.init(FakeMatcherConfig.builder()
+            .matcherName("PartHasContentType")
+            .condition("image/png")
+            .build());
+
+        Mail mail = FakeMail.builder()
+            .name("mail")
+            .recipient(ANY_AT_JAMES)
+            .mimeMessage(MimeMessageBuilder.mimeMessageBuilder()
+                .setMultipartWithBodyParts(
+                    MimeMessageBuilder.bodyPartBuilder()
+                        .type("text/plain")
+                        .disposition("attachment")
+                        .filename("xxx.zip")))
+            .build();
+        Collection<MailAddress> matched = testee.match(mail);
+
+        assertThat(matched)
+            .isNull();
+    }
+
+    @Test
+    void shouldMatchWhenGoodContentType() throws Exception {
+        PartHasContentType testee = new PartHasContentType();
+        testee.init(FakeMatcherConfig.builder()
+            .matcherName("PartHasContentType")
+            .condition("image/png")
+            .build());
+
+        Mail mail = FakeMail.builder()
+            .name("mail")
+            .recipient(ANY_AT_JAMES)
+            .mimeMessage(MimeMessageBuilder.mimeMessageBuilder()
+                .setMultipartWithBodyParts(
+                    MimeMessageBuilder.bodyPartBuilder()
+                        .type("image/png")
+                        .disposition("attachment")
+                        .filename("xxx.zip")))
+            .build();
+        Collection<MailAddress> matched = testee.match(mail);
+
+        assertThat(matched)
+            .isNotEmpty();
+    }
```

Comment:
How about a test with multiple content types configured with the matcher?

Reply:
That's the point of the refactoring: it leverages complexity of mime types and have that covered by AttachmentFileNameIs.

Code coverage is 100%

I do not see the point of adding more tests.